### PR TITLE
VIM-6206: Prevent Share Extension from Loading Main App's Descriptors

### PR DESCRIPTION
**Pull request by supperspidey**
#### Ticket

[VIM-6206](https://vimean.atlassian.net/browse/VIM-6206)

- *Note: this is required for Vimeo staff only. If not applicable to your PR, use "N/A".*

#### Pull Request Checklist

- [x] Resolved any merge conflicts
- [x] No build errors or warnings are introduced
- [x] New files are written in Swift
- [x] New classes contain license headers
- [x] New classes have Documentation
- [x] New public methods have Documentation

#### Issue Summary

This PR prevents the share extension from loading the main app's upload descriptors.

#### Implementation Summary

See [this PR](https://github.vimeows.com/MobileApps/Vimeo-iOS/pull/1984) for more details. 

#### Reviewer Tips

N/A

#### How to Test

- Confirm that both sample apps are still able to upload videos.

### CODE REVIEW

File: VimeoUpload/Upload/Controllers/VideoDeletionManager.swift [hunk 1]
```diff
@@ -106,14 +126,19 @@ public class VideoDeletionManager: NSObject
...
     }
     
     // MARK: Archiving
     
     private func loadDeletions() -> [VideoUri: Int]
     {
-        if let deletions = self.archiver?.loadObject(for: type(of: self).DeletionsArchiveKey) as? [VideoUri: Int]
+        guard self.shouldLoadArchive == true else
+        {
+            return [:]
```
Comment (jasonhawkins) [inline VimeoUpload/Upload/Controllers/VideoDeletionManager.swift]: Would it be worth it to just return nil here and change the method signature to return an optional?

File: VimeoUpload/Upload/Descriptor System/VideoDescriptorFailureTracker.swift [hunk 2]
```diff
@@ -86,17 +102,22 @@ import Foundation
...
-        return KeyedArchiver(basePath: typeFolderURL.path)
+        return KeyedArchiver(basePath: typeFolderURL.path, archivePrefix: archivePrefix)
     }
     
     private func load() -> [String: Descriptor]
     {
-        return self.archiver?.loadObject(for: type(of: self).ArchiveKey) as? [String: Descriptor] ?? [:]
+        guard self.shouldLoadArchive == true else
+        {
+            return [:]
```
Comment (jasonhawkins) [inline VimeoUpload/Upload/Descriptor System/VideoDescriptorFailureTracker.swift]: Same question here, regarding returning an optional instead?

**Comment by cameo-github** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          6 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoUpload/blob/7c0db11447ef097336d3f3fa06237b046c68ba8f/VimeoUpload/Upload/Descriptor System/Old Upload/OldUploadDescriptor.swift#L50">VimeoUpload/Upload/Descriptor System/Old Upload/OldUploadDescriptor.swift#L50</a></strong>: string interpolation produces a debug description for an optional value; did you mean to make this explicit?  <br /><code>
            print("\(self.currentRequest.rawValue) \(self.identifier)")
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoUpload/blob/7c0db11447ef097336d3f3fa06237b046c68ba8f/VimeoUpload/UIKit/DescriptorKVObserver.swift#L31">VimeoUpload/UIKit/DescriptorKVObserver.swift#L31</a></strong>: extraneous duplicate parameter name; ‘descriptorType’ already has an argument label  <br /><code>
    func descriptorStateDidChange(descriptorType descriptorType: VideoDescriptorType, descriptorState: DescriptorState, isCancelled: Bool, descriptorError: NSError?)
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoUpload/blob/7c0db11447ef097336d3f3fa06237b046c68ba8f/VimeoUpload/UIKit/DescriptorKVObserver.swift#L32">VimeoUpload/UIKit/DescriptorKVObserver.swift#L32</a></strong>: extraneous duplicate parameter name; ‘descriptorType’ already has an argument label  <br /><code>
    func descriptorProgressDidChange(descriptorType descriptorType: VideoDescriptorType, progress: Double)
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoUpload/blob/7c0db11447ef097336d3f3fa06237b046c68ba8f/VimeoUpload/Descriptor System/DescriptorManager.swift#L238">VimeoUpload/Descriptor System/DescriptorManager.swift#L238</a></strong>: conditional downcast from ‘Error?’ to ‘NSError’ is a bridging conversion; did you mean to use ‘as’?  <br /><code>
                let isConnectionError = ((task.error as? NSError)?.isConnectionError() == true || (error as? NSError)?.isConnectionError() == true)
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoUpload/blob/7c0db11447ef097336d3f3fa06237b046c68ba8f/VimeoUpload/Upload/Extensions/AVAsset+Extensions.swift#L32">VimeoUpload/Upload/Extensions/AVAsset+Extensions.swift#L32</a></strong>: extraneous duplicate parameter name; ‘completion’ already has an argument label  <br /><code>
    func approximateFileSize(completion completion: @escaping FloatBlock)
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoUpload/blob/7c0db11447ef097336d3f3fa06237b046c68ba8f/VimeoUpload/Upload/Descriptor System/Cameo/CAMUploadDescriptor.swift#L27">VimeoUpload/Upload/Descriptor System/Cameo/CAMUploadDescriptor.swift#L27</a></strong>: string interpolation produces a debug description for an optional value; did you mean to make this explicit?  <br /><code>
            print("\(self.currentRequest.rawValue) \(self.identifier)")
</code></td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          2 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 2 tests, with 0 failures (0 unexpected) in 0.909 (0.911) seconds</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 2 tests, with 0 failures (0 unexpected) in 0.342 (0.344) seconds</td>
    </tr>
  </tbody>
</table>

See build details on [CircleCI](https://circleci.com/gh/vimeo/VimeoUpload/310)

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by supperspidey** [comment]
Thanks @jasonhawkins !

**Pull request merged**